src/devices/PiJuice/PiJuice.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.I2c;
+using System.IO;
+using System.Linq;
+using Iot.Device.PiJuiceDevice.Models;
+
+namespace Iot.Device.PiJuiceDevice
+{
+    /// <summary>
+    /// Create a PiJuice class
+    /// </summary>
+    public class PiJuice : IDisposable
+    {
+        private const byte MaxRetries = 4;
+        private const int ReadRetryDelay = 50;
+
+        private readonly bool _shouldDispose;
+
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// The default PiJuice I2C address is 0x14
+        /// Other addresses can be uses, see PiJuice documentation
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x14;
+
+        /// <summary>
+        /// Contains the PiJuice key information
+        /// </summary>
+        public PiJuiceInfo PiJuiceInfo { get; internal set; }
+
+        /// <summary>Creates a new instance of the PiJuice.</summary>
+        /// <param name="i2cDevice">The I2C device. Device address is 0x14</param>
+        /// <param name="shouldDispose">True to dispose the I2C device</param>
+        public PiJuice(I2cDevice i2cDevice, bool shouldDispose = true)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _shouldDispose = shouldDispose;
+            PiJuiceInfo = new PiJuiceInfo() { FirmwareVersion = GetFirmwareVerion() };
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            if (_shouldDispose)
+            {
+                _i2cDevice?.Dispose();
+                _i2cDevice = null;
+            }
+        }
+
+        /// <summary>
+        /// Get the firmware version
+        /// </summary>
+        /// <returns>PiJuice firmware version</returns>
+        public Version GetFirmwareVerion()
+        {
+            var response = ReadCommand(PiJuiceCommand.FirmwareVersion, 2);
+
+            var major_version = response[0] >> 4;
+            var minor_version = (response[0] << 4 & 0xf0) >> 4;
+
+            return new Version(major_version, minor_version);
+        }
+
+        /// <summary>
+        /// Restore PiJuice to its default settings
+        /// </summary>
+        public void SetDefaultConfiguration()
+        {
+            WriteCommand(PiJuiceCommand.ResetToDefault, new byte[] { 0xAA, 0x55, 0x0A, 0xA3 });
+        }
+
+        /// <summary>
+        /// Get the write protect of the EEPROM
+        /// </summary>
+        /// <returns>Whether the EEPROM is write protected</returns>
+        public bool GetIdEepromWriteProtect()
```

Comment:
You could make the `IdEepromWriteProtect` a property. There is a set and a get, it doesn't change over time